test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,43 @@
+require "test_helper"
+
+describe "UsersController" do
+  describe "login_form" do
+    it "can render the login form" do
+      get login_path
+      must_respond_with :ok
+    end
+  end
+
+  describe "login" do
+    it "logs in an user" do
+      new_user = User.create(username: "newuser")
+
```

Comment:
There are a couple of behavior modes you're not covering:
- Logging in a new user (should create that user)
- Attempting to log in with an invalid username (e.g. empty string) (should fail)